.github/proverCiScripts/injectHalo2Revision.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+#set -e 
+#set -x
+
+prnumber=$1
+base_dir="/home/ubuntu/CI_Prover_Benches/"
+target_dir="$base_dir"PR"$1"
+
+_revision=$2
+
+echo $_revision
+
+oldrev=$(egrep halo2.*rev $target_dir/Cargo.toml | awk -F\" '{ print $4 }')
+
+sed -i "s/$oldrev/$_revision/g" $target_dir/Cargo.toml
```

Comment:
The halo2 revision is also specified in these `Cargo.toml` files:
- `keccak256/Cargo.toml`
- `zkevm-circuits/Cargo.toml`
- `integration-tests/Cargo.toml`
- `circuit-benchmarks/Cargo.toml`

I think those should be updated as well.

The other option I can think of changing the `Cargo.toml` files inside of each workspace member so that it doesn't specify the rev, so that they pick the one defined in the root `Cargo.toml`.  Although I'm not 100% sure this would work.

Reply:
I should be on the workspace one to no? Isn't that one valid?

Reply:
Thanks Edu, i am adding a simple iteration to work around this issue: we will read the root Cargo.toml and extract the halo2 revision previously used. Then, we do in place replacement (happens only on the checked out code in prover node) of the hash string in all Cargo.toml files where it exists with the halo2 PR head sha received